src/SDK/SdkAutoloader.php

```diff
@@ -30,80 +36,121 @@ public static function autoload(): bool
+
+        ShutdownHandler::register($tracerProvider->shutdown(...));
+        ShutdownHandler::register($meterProvider->shutdown(...));
+        ShutdownHandler::register($loggerProvider->shutdown(...));
+
+        return $configurator
+            ->withTracerProvider($tracerProvider)
+            ->withMeterProvider($meterProvider)
+            ->withLoggerProvider($loggerProvider)
+            ->withEventLoggerProvider($eventLoggerProvider)
+            ->withPropagator($propagator)
+        ;
+    }
+
+    private static function createSdkFromFileConfig(): Sdk
+    {
+        $file = Configuration::getString(Variables::OTEL_EXPERIMENTAL_CONFIG_FILE);
+        $config = SdkConfiguration::parseFile($file);
+
+        return $config
+            ->create()
+            ->setAutoShutdown(true)
+            ->build();
+    }
+
     /**
-     * Test whether a request URI is set, and if it matches the excluded urls configuration option
+     * Register any auto-instrumentation configured through SPI
      *
-     * @internal
+     * @phan-suppress PhanUndeclaredClassMethod
      */
-    public static function isIgnoredUrl(): bool
+    private static function registerInstrumentations(?Sdk $sdk): void
     {
-        $ignoreUrls = Configuration::getList(Variables::OTEL_PHP_EXCLUDED_URLS, []);
-        if ($ignoreUrls === []) {
-            return false;
+        if (!Configuration::has(Variables::OTEL_PHP_INSTRUMENTATION_CONFIG_FILE)) {
+            return;
         }
-        $url = $_SERVER['REQUEST_URI'] ?? null;
-        if (!$url) {
-            return false;
+        $file = Configuration::getString(Variables::OTEL_PHP_INSTRUMENTATION_CONFIG_FILE);
+        $configuration = Instrumentation::parseFile($file)->create();
+        $storage = Context::storage();
+        $hookManager = self::getHookManager();
+        $context = new InstrumentationContext(
+            $sdk?->getTracerProvider() ?? Globals::tracerProvider(),
+            $sdk?->getMeterProvider() ?? Globals::meterProvider(),
+            $sdk?->getLoggerProvider() ?? Globals::loggerProvider(),
+        );
+        foreach (ServiceLoader::load(\OpenTelemetry\API\Instrumentation\AutoInstrumentation\Instrumentation::class) as $instrumentation) {
+            /** @var \OpenTelemetry\API\Instrumentation\AutoInstrumentation\Instrumentation $instrumentation */
+            $instrumentation->register($hookManager, $context, $configuration, $storage);
         }
-        foreach ($ignoreUrls as $ignore) {
-            if (preg_match(sprintf('|%s|', $ignore), (string) $url) === 1) {
-                return true;
-            }
+    }
+
+    /**
+     * @phan-suppress PhanUndeclaredClassMethod
+     */
+    private static function getHookManager(): HookManager
+    {
+        /** @var HookManager $hookManager */
+        foreach (ServiceLoader::load(HookManager::class) as $hookManager) {
+            $scope = $hookManager->enable(Context::getCurrent())->activate();
```

Comment:
> We could add a method that allows changing hook managers from default disabled to default enabled which would make this scope obsolete.

I do think it would be preferable if the hook manager was enabled by default. If you've installed auto-instrumentation packages, you should get auto-instrumentation (which is the current behaviour). Plus, I don't want developers to need to write code to enable hooks to run; the most common usage I think would be to instrument.

Reply:
I've switched `ExtensionHookManager` to being enabled by default. How one might access the hook manager to disable it isn't very clear if it was autoloaded - maybe it needs to be a singleton or registered in context?